src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdLoadBarrier.inline.hpp

```diff
@@ -65,12 +65,16 @@ inline traceid set_used_and_get(const T* type) {
   return TRACE_ID(type);
 }
 
-inline traceid JfrTraceIdLoadBarrier::load(const Klass* klass) {
-  assert(klass != NULL, "invariant");
-  if (should_tag(klass)) {
+inline void JfrTraceIdLoadBarrier::load_barrier(const Klass* klass) {
     SET_USED_THIS_EPOCH(klass);
     enqueue(klass);
     JfrTraceIdEpoch::set_changed_tag_state();
```

Comment:
JfrTraceIdEpoch::set_changed_tag_state(); here now obviates the need to program signal into the intrinsic so that can be removed. And signal_address() need not be exposed.

Reply:
`load_barrier` is used by the path for normal class and `JfrTraceIdLoadBarrier::load(const Klass* klass)`
the path for the primitive class still needs to update the signal.

Reply:
ah..ok...